operations/kubernetes/Determine_if_Pods_are_Hitting_Resource_Limits.md

```diff
@@ -1,134 +1,106 @@
+       ```bash
+       /opt/cray/platform-utils/detect_cpu_throttling.sh externaldns
+       ```
 
-        ```bash
-        ncn-w001# cat detect_cpu_throttling.sh
-        #!/bin/sh
-        # Usage: detect_cpu_throttling.sh [pod_name_substr] (default evaluates all pods)
+       Example output:
 
-        str=$1
-        : ${str:=.}
+       ```text
+       Checking cray-externaldns-coredns-58b5f8494-c45kh
 
-        while read ns pod node; do
-          echo ""
-          echo "Checking $pod"
-          while read -r container; do
-            uid=$(echo $container | awk 'BEGIN { FS = "/" } ; {print $NF}')
-            ssh -T ${node} <<-EOF
-                dir=$(find /sys/fs/cgroup/cpu,cpuacct/kubepods/burstable -name *${uid}* 2>/dev/null)
-                [ "${dir}" = "" ] && { dir=$(find /sys/fs/cgroup/cpu,cpuacct/system.slice/containerd.service -name *${uid}* 2>/dev/null); }
-                if [ "${dir}" != "" ]; then
-                  num_periods=$(grep nr_throttled ${dir}/cpu.stat | awk '{print $NF}')
-                  if [ ${num_periods} -gt 0 ]; then
-                    echo "*** CPU throttling for containerid ${uid}: ***"
-                    cat ${dir}/cpu.stat
-                    echo ""
-                  fi
-                fi
-        	EOF
-          done <<< "`kubectl -n $ns get pod $pod -o yaml | grep ' - containerID'`"
-        done <<<"$(kubectl get pods -A -o wide | grep $str | grep Running | awk '{print $1 " " $2 " " $8}')"
-        ```
+       Checking cray-externaldns-coredns-58b5f8494-pjvz6
 
-    2.  Determine if any pods are being CPU throttled.
+       Checking cray-externaldns-etcd-2kn7w6gnsx
 
-        The script can be used in two different ways:
+       Checking cray-externaldns-etcd-88x4drpv27
 
-    -   Pass in a substring of the desired pod name\(s\).
+       Checking cray-externaldns-etcd-sbnbph52vh
 
-        In the example below, the externaldns pods are being used.
+       Checking cray-externaldns-external-dns-5bb8765896-w87wb
+       *** CPU throttling: ***
+       nr_periods 1127304
+       nr_throttled 473554
+       throttled_time 71962850825439
+       ```
 
-        ```bash
-        ncn-w001# ./detect_cpu_throttling.sh externaldns
-        ```
+    1. (`ncn-mw#`) Call the script without a parameter to evaluate all pods.
 
-        Example output:
+       This can take two minutes or more to run when evaluating all pods:
 
-        ```
-        Checking cray-externaldns-coredns-58b5f8494-c45kh
+       ```bash
+       /opt/cray/platform-utils/detect_cpu_throttling.sh
+       ```
 
-        Checking cray-externaldns-coredns-58b5f8494-pjvz6
+       Example output:
 
-        Checking cray-externaldns-etcd-2kn7w6gnsx
+       ```text
+       Checking benji-k8s-fsfreeze-9zlfk
 
-        Checking cray-externaldns-etcd-88x4drpv27
+       Checking benji-k8s-fsfreeze-fgqmd
 
-        Checking cray-externaldns-etcd-sbnbph52vh
+       Checking benji-k8s-fsfreeze-qgbcp
```

Comment:
```suggestion
```

Reply:
would you rather see pods other than benji?  formatting looks ok..